src/common/hotbar-store.ts

```diff
@@ -192,6 +192,39 @@ export class HotbarStore extends BaseStore<HotbarStoreModel> {
     hotbar.items[index] = null;
   }
 
+  /**
+   * Remvove all hotbar items that reference the `uid`.
+   * @param uid The `EntityId` that each hotbar item refers to
+   * @returns A function that will (in an action) undo the removing of the hotbar items. This function will not complete if the hotbar has changed.
+   */
+  @action
+  removeAllHotbarItems(uid: string): UndoFunction {
+    const undoItems: [Hotbar, number, HotbarItem][] = [];
+
+    for (const hotbar of this.hotbars) {
+      const index = hotbar.items.findIndex((i) => i?.entity.uid === uid);
+
+      if (index >= 0) {
+        undoItems.push([hotbar, index, hotbar.items[index]]);
+        hotbar.items[index] = null;
+      }
+    }
+
+    return () => runInAction(() => {
+      for (const [hotbar, index] of undoItems) {
+        if (hotbar.items[index] === null) {
```

Comment:
Did you mean `!==`?
Do we even need an undo? I think you're trying to do too much. If the delete fails for some reason it might be better to just leave the hotbar alone. It's ok that the cluster is no longer in the hotbar(s)

Reply:
okay, I'll remove this. I just wanted the hotbar items to be still there if (for some reason) we failed due to missing write permissions

Reply:
yeah, it's a commendable idea, but on the other hand the user asked us to remove the cluster, and taking it out of the hotbar(s) could be considered at least a partial removal...

Reply:
Fair, I was treating it more of a DB transaction where if something fails we try and back out of all the changes. This is probably too complex though.